dashboard/app/assets/javascripts/progress.js

```diff
@@ -76,15 +76,19 @@ window.dashboard.progress = (function () {
     }
   };
 
-  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId) {
+  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId, puzzlePage) {
     var serverProgress = progressData.levels || {};
     var currentLevelIndex = null;
 
-    var combinedProgress = stageData.levels.map(function(level, index) {
-      if (level.id === currentLevelId) {
-        currentLevelIndex = index;
-      }
+    // Determine currentLevelIndex from the URL's /puzzle/XX value.
+    var parsedLevelIndex = window.location.pathname.match(/\/puzzle\/(\d+)/);
```

Comment:
I think easiest to just go back to earlier implementation.  It also works with the /hoc/1 format.